core/src/main/java/hudson/Launcher.java

```diff
@@ -761,6 +761,7 @@ protected final void printCommandLine(@Nonnull String[] cmd, @CheckForNull FileP
                 buf.append(c);
         }
         listener.getLogger().println(buf.toString());
+        listener.getLogger().flush();
```

Comment:
If this is commented out, `ArgumentListBuilder2Test.slaveMask` fails (gets a blank string)—unless you start the test with

```java
static {
    System.setProperty("hudson.util.StreamTaskListener.AUTO_FLUSH", "true");
}
```

in which case it passes, demonstrating the effectiveness of the flag on the remote side.